.github/workflows/python-publish-release.yml

```diff
@@ -69,7 +69,7 @@ jobs:
   publish:
     needs: [build_wheels]
     runs-on: ubuntu-latest
-    if: startsWith(github.ref, 'refs/tags')
+    if: github.event_name == 'release' && github.event.prerelease == false && github.event.action == 'published'
```

Comment:
Why we need change this line? Is it tested?

Reply:
We don't need to. But the previous one will trigger the upload on every tag. Which is not what we want necessarily(?). We want to upload the wheels on every release, right?
Because currently we can't create a new tag without also uploading the wheels.
But we might want to create tags which are no release.

Reply:
I copied the line from here ad extended it to check for pre-releases:
https://github.com/pypa/cibuildwheel/blob/170e7aaeb91807dccbade786f95e8c8e4d2eda31/examples/github-deploy.yml#L51-L56

Reply:
Here the test:

Triggered on pre-release. Should not execute the publish/upload step:

https://github.com/Rot127/capstone/actions/runs/10447395841

Reply:
@doronz88 You can see in https://github.com/Rot127/capstone/actions/runs/10447395841/job/28926636617 that in the download artifact step it gets all artifacts. Not just one.

Reply:
See testing procedure described in main description